src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -172,6 +181,10 @@ static void showSettings(boolean printToStderr, String optionFlag,
             case "locale":
                 printLocale();
                 break;
```

Comment:
printLocale might need the same treatment to avoid printing the list of all locales by default.

Reply:
Have you tried putting the new code into say sun.launcher.SecuritySettings to avoid adding so much security conf code into LauncherHelper? Maybe eventually this will need to a more pluggable as -XshowSettings has grown a bit beyond what it was originally for.

Reply:
Logged https://bugs.openjdk.org/browse/JDK-8310201 to track the Locale changes Alan.

Good suggestion on factoring out the security code to a helper class. Done.

Reply:
> Logged https://bugs.openjdk.org/browse/JDK-8310201 to track the Locale changes Alan.

Thanks, probably should treat this as a bug for JDK 21 so that a new addition doesn't change behavior between 21 and 22.